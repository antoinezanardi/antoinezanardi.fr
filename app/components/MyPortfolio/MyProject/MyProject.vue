<template>
  <div
    class="cc-portfolio-image img-raised"
    data-aos="fade-up"
    data-aos-anchor-placement="top-bottom"
  >
    <a
      class="project-link"
      :href="project.url"
      rel="noopener noreferrer"
      target="_blank"
    >
      <figure class="cc-effect">
        <NuxtImg
          :alt="project.name"
          class="project-image"
          loading="lazy"
          :src="`/images/projects/${project.image}`"
        />

        <figcaption class="d-flex flex-column h-100 justify-content-center project-metadata">
          <div
            class="h4 project-name py-3"
          >
            {{ project.name }}
          </div>

          <p
            class="align-items-center d-flex flex-column justify-content-center project-desc"
          >
            <span>
              {{ project.description }}
            </span>

            <WrappedFontAwesomeIcon
              classes="my-3"
              icon="fa-arrow-up-right-from-square"
              icon-color="#000000"
              size="2x"
            />
          </p>
        </figcaption>
      </figure>
    </a>
  </div>
</template>

<script setup lang="ts">
import type { MyProjectProps } from "~/components/MyPortfolio/MyProject/my-project.types";
import WrappedFontAwesomeIcon from "~/components/shared/Icons/WrappedFontAwesomeIcon/WrappedFontAwesomeIcon.vue";

defineProps<MyProjectProps>();
</script>

<style lang="scss" scoped>
.project-metadata {
  text-shadow: 2px 2px 15px black;
}
</style>