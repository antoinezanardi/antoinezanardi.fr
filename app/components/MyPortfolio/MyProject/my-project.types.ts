import type { Project } from "~/models/Project";

type MyProjectProps = {
  project: Project;
};

export type { MyProjectProps };