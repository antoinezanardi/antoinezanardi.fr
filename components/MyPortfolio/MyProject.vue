<template>
  <div class="cc-porfolio-image img-raised" data-aos="fade-up" data-aos-anchor-placement="top-bottom">
    <a target="_blank" rel="noopener noreferrer" :href="project.url">
      <figure class="cc-effect">
        <NuxtImg :src="`/images/projects/${project.image}`" format="webp" :alt="project.name"/>
        <figcaption class="project-metadata">
          <div class="h4" v-html="project.name"/>
          <p class="project-desc" v-html="project.description"/>
        </figcaption>
      </figure>
    </a>
  </div>
</template>

<script setup lang="ts">
import type { Project } from "~/models/Project";

defineProps<{
  project: Project;
}>();
</script>

<style lang="scss" scoped>
.project-metadata {
  text-shadow: 2px 2px 15px black;
}
</style>