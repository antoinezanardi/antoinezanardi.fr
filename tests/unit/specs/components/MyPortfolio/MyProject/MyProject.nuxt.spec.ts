import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";
import MyProject from "~/components/MyPortfolio/MyProject/MyProject.vue";
import type { MyProjectProps } from "~/components/MyPortfolio/MyProject/my-project.types";
import type { NuxtImg } from "#components";

describe("My Project Component", () => {
  const defaultProps: MyProjectProps = {
    project: {
      name: "MyPortfolio.projects.portfolio.name",
      description: "MyPortfolio.projects.portfolio.description",
      image: "portfolio-thumbnail.webp",
      url: "https://www.antoinezanardi.fr",
    },
  };
  let wrapper: ReturnType<typeof mount<typeof MyProject>>;

  async function mountMyProjectComponent(options: ComponentMountingOptions<typeof MyProject> = {}): Promise<ReturnType<typeof mount<typeof MyProject>>> {
    return mountSuspendedComponent(MyProject, {
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountMyProjectComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Image", () => {
    it("should set alt to project name when rendered.", () => {
      const image = wrapper.findComponent<typeof NuxtImg>(".project-image");

      expect(image.attributes("alt")).toBe(defaultProps.project.name);
    });

    it("should set image source when rendered.", () => {
      const image = wrapper.findComponent<typeof NuxtImg>(".project-image");

      expect(image.props("src")).toBe("/images/projects/portfolio-thumbnail.webp");
    });
  });

  describe("Project", () => {
    it("should render project name when rendered.", () => {
      const projectName = wrapper.find<HTMLHeadingElement>(".project-name");

      expect(projectName.text()).toBe("MyPortfolio.projects.portfolio.name");
    });

    it("should render project description when rendered.", () => {
      const projectDescription = wrapper.find<HTMLParagraphElement>(".project-desc");

      expect(projectDescription.text()).toBe("MyPortfolio.projects.portfolio.description");
    });
  });
});