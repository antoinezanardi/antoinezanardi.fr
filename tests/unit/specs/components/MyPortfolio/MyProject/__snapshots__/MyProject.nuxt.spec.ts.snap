// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`My Project Component > should match snapshot when rendered. 1`] = `
"<div data-v-09377792="" class="cc-portfolio-image img-raised" data-aos="fade-up" data-aos-anchor-placement="top-bottom"><a data-v-09377792="" class="project-link" href="https://www.antoinezanardi.fr" rel="noopener noreferrer" target="_blank">
    <figure data-v-09377792="" class="cc-effect">
      <nuxt-img-stub data-v-09377792="" alt="MyPortfolio.projects.portfolio.name" loading="lazy" src="/images/projects/portfolio-thumbnail.webp" preload="false" crossorigin="false" placeholder="false" custom="false" class="project-image"></nuxt-img-stub>
      <figcaption data-v-09377792="" class="d-flex flex-column h-100 justify-content-center project-metadata">
        <div data-v-09377792="" class="h4 project-name py-3">MyPortfolio.projects.portfolio.name</div>
        <p data-v-09377792="" class="align-items-center d-flex flex-column justify-content-center project-desc"><span data-v-09377792="">MyPortfolio.projects.portfolio.description</span>
          <wrapped-font-awesome-icon-stub data-v-09377792="" icon="fa-arrow-up-right-from-square" iconcolor="#000000" size="2x" classes="my-3"></wrapped-font-awesome-icon-stub>
        </p>
      </figcaption>
    </figure>
  </a></div>"
`;
